### PR TITLE
[Concurrency] Simple Task.yield implementation

### PR DESCRIPTION
**Pull request by ktoso**
Pretty simple/naive implementation of Task.yield...

I'm guessing this isn't quite the final form of this.

It is the same impl style as today's Task.sleep: https://github.com/apple/swift/pull/35983

Though I think we should somehow implement be able to implement this without creating a new Task.

### ABI

- This likely should change to have a single runtime call to a new `swift_task_yield`?

### CODE REVIEW

File: stdlib/public/Concurrency/Task.swift [hunk 1]
```diff
@@ -472,6 +472,38 @@ public func _runAsyncHandler(operation: @escaping () async -> ()) {
...
+    // FIXME: This should be an empty closure instead. Returning `0` here is
+    //        a workaround for rdar://74957357
+    // TODO: consider if it would be useful for this task to be a child task
+    let (task, _) = Builtin.createAsyncTaskFuture(flags.bits, nil, { return 0 })
+
+    // Enqueue the resulting job.
+    _enqueueJobGlobal(Builtin.convertTaskToJob(task))
+
+    let _ = await Handle<Int, Never>(task).get()
+  }
```
Comment (ktoso) [inline stdlib/public/Concurrency/Task.swift:503]: So this is a pretty naive implementation;

As with sleep() I am wondering if we really need to create a new task to do this?

See also: https://github.com/apple/swift/pull/35983#discussion_r576530459

File: test/Concurrency/Runtime/async_task_yield.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  }
+}
+
+@main struct Main {
+  static func main() async {
+    await yielding()
+    // TODO: No idea for a good test for this... Open to ideas?
+    // CHECK:      One
+    // CHECK-NEXT: Two
+  }
```
Comment (ktoso) [inline test/Concurrency/Runtime/async_task_yield.swift:41]: no idea how to write a good test for this without a custom executor to inspect the tasks submitted 🤔

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: stdlib/public/Concurrency/Task.swift (hunk 1 above)
Comment (ktoso) [reply stdlib/public/Concurrency/Task.swift:529]: (I'm looking into implementing this without a new Task now)

**Comment by CodaFi** [comment]
@swift-ci smoke test Linux

**Comment by ktoso** [comment]
Heh linux execution actually ends up completely serial... even with the yields which I had hoped will be enough even on the single threaded execution context...

```
06:02:55         97: One @ 96
06:02:55         98: One @ 97
06:02:55         99: One @ 98
06:02:55        100: One @ 99
06:02:55        101: One @ 100
06:02:55        102: Two @ 0
06:02:55 next:40     !~~     error: match on wrong line
06:02:55        103: Two @ 1
06:02:55        104: Two @ 2
06:02:55        105: Two @ 3
06:02:55        106: Two @ 4
06:02:55        107: Two @ 5
```

Need to think if this is because the cooperative scheduling is crappy or the yield is not really doing enough 🤔 

Maybe you'd have some insight @drexin ?

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Review by drexin — APPROVED** [review]
LGTM

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test macos

**Comment by drexin** [comment]
@swift-ci smoke test windows

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Pull request merged**